### ISSUE TITLE
Frequent front-end crash on alert list page with 

### ISSUE THREAD

**Issue by adeverteuil** [grafana/grafana#57206]
**What happened**:

I'm working on a Grafana Cloud stack that has 1612 alert rules in Loki and Prometheus (and I'm about to upload ~15K more rules to Loki). When I look at the Alerts rules page, the new pagination system makes it much more manageable than it used to be. However, quite frequently the page crashes with an "An unexpected error happened".

![Screenshot 2022-10-18 13-57-30 crash](https://user-images.githubusercontent.com/3270755/196508744-e94770f8-3acc-4f2b-93f7-92bc6e0c289e.png)

Here is the text content of the error details:

```
TypeError: t.map is not a function

81356/yr<@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/AlertRuleListIndex.2ae4324fc2c8e66cb1b2.js:152:4700
c@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:1194:301
u@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:1194:855
WithErrorBoundary
Suspense
c@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:1194:301
kn@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:5515:360
t@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/5290.5730ffe229a3937f6f65.js:2:936126
t@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/5290.5730ffe229a3937f6f65.js:2:938147
main
st@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:5476:13057
t@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/5290.5730ffe229a3937f6f65.js:2:932744
div
o@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:2035:4592
19175/t.KBarProvider@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/5290.5730ffe229a3937f6f65.js:2:182746
l@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:6237:19191
c@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:1194:301
u@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:1194:855
y@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/5290.5730ffe229a3937f6f65.js:2:2813673
u@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:6230:10878
l@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/5290.5730ffe229a3937f6f65.js:2:918851
ss@https://grafana-assets.grafana.net/grafana-pro/9.2.0/public/build/3191.e394a9398e295f847fb0.js:5528:167
```

Here is the error message copied from the web console in my browser:

```
TypeError: t.map is not a function
    i useCombinedRuleNamespaces.ts:119
    i useCombinedRuleNamespaces.ts:72
    i useCombinedRuleNamespaces.ts:65
    i useCombinedRuleNamespaces.ts:53
    React 2
    i useCombinedRuleNamespaces.ts:50
    yr RuleList.tsx:77
react-dom.production.min.js:216:199
    React 5
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    Redux 33
    W actions.ts:309
    W actions.ts:295
    a Redux
    r RuleList.tsx:54
    (Async: setInterval handler)
    yr RuleList.tsx:54
    Nu React
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    unstable_runWithPriority scheduler.production.min.js:18
    React 2
    I scheduler.production.min.js:16
    onmessage scheduler.production.min.js:12
    (Async: EventHandlerNonNull)
    15303 scheduler.production.min.js:12
    Webpack 26

```

I can refresh the page and continue working.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

1. I think you need an instance with a Loki or Prometheus data source with > 1000 alerts. Not sure at what scale the problem happens.
2. Open the Alert rules page.
3. Wait.

**Anything else we need to know?**:

**Environment**:
- Grafana version: v9.2.0
- Data source type & version: Grafana Cloud Loki and Prometheus
- OS Grafana is installed on: Grafana Cloud
- User OS & Browser: Firefox 105.0.1 on Pop_OS 22.04 (based on Ubuntu 22.04)
- Grafana plugins:
- Others:

**Comment by alexweav** [comment]
There's an incident open - #incident-2022-10-14-multiple-customers-see-alert-list-crashes

this impacts the entirety of Grafana, any view that sends concurrent requests to different endpoints is subject to this issue

**Comment by joeblubaugh** [comment]
We've been able to reproduce this issue and are working to find the exact cause.

**Comment by jaygridley** [comment]
Any progress with this issue?

**Comment by joeblubaugh** [comment]
Apologies for the silence on this issue. In investigating it we found some significant security problems across Grafana.

The fix has been released in a few different versions, the most recent one being v9.2.4.

For more information, see:
https://grafana.com/blog/2022/11/08/security-release-new-versions-of-grafana-with-critical-and-moderate-fixes-for-cve-2022-39328-cve-2022-39307-and-cve-2022-39306/

This was CVE-2022-39328
